### DISCUSSION TITLE
What can we do to bypass the Godot color tolerance bug?

### DISCUSSION THREAD

**Discussion by BrotherShort** [Orama-Interactive/Pixelorama#1158] [Ideas]
#1155 reported a color tolerance bug. 
The bug is from Godot. There is a Godot issue about it. https://github.com/godotengine/godot/issues/59983 It won't be fixed by Godot soon.
Maybe for games, it's not a too bad bug, but for image processing software, it's damaging.
Can Pixelorama do something to bypass the bug? Pixelorama always uses FORMAT_RGBA8 as image format. I think we could use FORMAT_RGBAF instead of FORMAT_RGBA8. FORMAT_RGBAF won't <del>departure</del> trigger the bug.

**Comment by OverloadedOrama** [comment]
If the bug doesn't occur when using `FORMAT_RGBAF` it might be our best bet. I wonder if there are any drawbacks to performance however. From what I see in the documentation, each channel has 32-bit floating point values, compared to RGBA8's 8-bit. I assume this would result in RGBAF needing more memory? Perhaps if there is no other drawback we could have `FORMAT_RGBAF` by default and let users change to RGBA8 if they want to save memory.
